nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/iot/util/MqttWebSocketAsyncClient.java

```diff
@@ -0,0 +1,113 @@
+            return original;
+        }
+        final URI uri = URI.create(original);
+        return "tcp://DUMMY-" + uri.getHost() + ":"
+                + (uri.getPort() > 0 ? uri.getPort() : 80);
+    }
+
+    protected static boolean isDummyURI(String uri) {
+        return uri.startsWith("tcp://DUMMY-");
+    }
+
+    public MqttWebSocketAsyncClient(String serverURI, String clientId,
+                                    ProcessorLog logger) throws MqttException {
+        super(createDummyURI(serverURI), clientId, new MemoryPersistence(), new TimerPingSender());
+        this.serverURI = serverURI;
+        this.logger = logger;
+        this.setCallback(this);
+    }
+
+    @Override
+    protected NetworkModule[] createNetworkModules(String address,
+                                                   MqttConnectOptions options) throws MqttException{
+        String[] serverURIs = options.getServerURIs();
+        String[] array = serverURIs == null ? new String[] { address } :
+            serverURIs.length == 0 ? new String[] { address }: serverURIs;
+
+        NetworkModule[] networkModules = new NetworkModule[array.length];
+        for (int i = 0; i < array.length; i++) {
+            networkModules[i] = createNetworkModule(array[i], options);
+        }
+        return networkModules;
+    }
+
+    protected NetworkModule createNetworkModule(String input,
+                                                MqttConnectOptions options) throws MqttException,
+            MqttSecurityException {
+        final String address = isDummyURI(input) ? this.serverURI : input;
+        if (!address.startsWith("ws:") && !address.startsWith("wss:")) {
+            return super.createNetworkModules(address, options)[0];
+        }
+
+        final String subProtocol = (options.getMqttVersion() == MqttConnectOptions.MQTT_VERSION_3_1) ? "mqttv3.1" : "mqtt";
+        return newWebSocketNetworkModule(URI.create(address), subProtocol, options);
+    }
+
+    protected NetworkModule newWebSocketNetworkModule(URI uri,
+                                                      String subProtocol, MqttConnectOptions options) {
+        final WebSocketNetworkModule netModule = new WebSocketNetworkModule(
+                uri, subProtocol, getClientId());
+        netModule.setConnectTimeout(options.getConnectionTimeout());
+        return netModule;
+    }
+
+    public LinkedBlockingQueue<IoTMessage> getAwsQueuedMqttMessages() {
+        return awsQueuedMqttMessages;
+    }
+
+    @Override
+    public void connectionLost(Throwable t) {
+        logger.error("Connection to " + this.getServerURI() + " lost with cause: " + t.getMessage());
+    }
+
+    @Override
+    public void deliveryComplete(IMqttDeliveryToken token) {
+    }
+
+    @Override
+    public void messageArrived(String topic, MqttMessage message) throws Exception {
+        logger.info("Message arrived from topic: " + topic);
```

Comment:
This came up in my review as well (I did this at first), we shouldn't set the messageArrived message to INFO. The rate at which MQTT message potentially arrive would fill up the logs very quickly. Check out how I handle it in ConsumeMQTT[1].

[1] https://github.com/apache/nifi/blob/f47af1ce8336c9305916f00738976f3505b01b0b/nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/ConsumeMQTT.java#L310